File: algorithms/2_medium/L/03702_longest_subsequence_with_non_zero_bitwise_xor/solution.c
int longestSubsequence(const int *restrict nums, const int len) {
    if (len == 0 || nums == NULL) { return 0; }
    int xor = 0;
    int zeros = 0;

    for (int i = 0; i < len; i += 1) {
        xor ^= nums[i];
        if (nums[i] == 0) {
            zeros += 1;
        }
    }

    if (zeros == len) {
        return 0;
    }

    if (xor != 0) {
        return len;
    }

    return len - 1;
}

File: algorithms/2_medium/L/03702_longest_subsequence_with_non_zero_bitwise_xor/solution.rs
impl Solution {
    pub fn longest_subsequence(nums: Vec<i32>) -> i32 {
        let len = nums.len();
        let mut xor = 0_i32;
        let mut zeros = 0_usize;

        for num in nums {
            xor ^= num;
            if num == 0 {
                zeros += 1;
            } 
        }

        if zeros == len {
            return 0;
        }

        if xor != 0 {
            return len as i32;
        }

        len as i32 - 1
    }
}
